.github/workflows/cmake.yml

```diff
@@ -10,9 +10,9 @@ jobs:
       fail-fast: false
       matrix:
         build_type : [ Release, Debug ]
-        os : [ macos-latest, ubuntu-20.04 ]
+        os : [ macos-latest, ubuntu-22.04 ]
```

Comment:
We can also stick with 20.04 and install Boost externally instead of via the Ubuntu repos. Upgrading the CI just seemed to be the simplest solution.

Reply:
What's up the Ubuntu repo's Boost version in 20.04?

Reply:
It's too old (1.71, but we need 1.72). See https://packages.ubuntu.com/focal/libboost-dev

Reply:
The parser you wrote requires at least 1.72?

Reply:
Yes - that's why I increased the minimum Boost version to 1.72. Or more specifically: prior Boost versions contain a bug (https://github.com/boostorg/spirit/issues/533) that makes the grammar fail to compile and this bug was fixed in Boost 1.72

Reply:
I suggest we hard-code the fix of that issue in SeQuant whenever the Boost version is <1.72.
The issue is fixed [in this thread](https://github.com/boostorg/spirit/issues/533#issuecomment-530197276).

Setting the minimum Boost version to 1.72 in SeQuant also requires doing the same in mpqc4 (and any other programs that use SeQuant).

@Krzmbrzl

Reply:
That seems very error prone and not future-proof. Also I don't see why the other lobs would also have to increase their minimum Boost version? If SeQuant is used somewhere, the minimum version will be dictated by it and otherwise the newer versions are not need :thinking: